util/syspolicy/syspolicy_test.go

```diff
@@ -67,13 +74,246 @@ func TestGetString(t *testing.T) {
 			if value != tt.value {
 				t.Fatalf("got value %v instead of expected value %v", value, tt.value)
 			}
-			if err != tt.err {
+			if err != tt.err && tt.err != ErrNoSuchKey {
 				t.Fatalf("got error %v instead of expected error %v", err, tt.err)
 			}
 		})
 	}
 }
 
+func TestGetUint64(t *testing.T) {
+	tests := []testHandler{
+		{
+			t:    t,
+			name: "read existing value",
+			key:  KeyExpirationNoticeTime,
+			u64:  1,
+			err:  nil,
+		},
+		{
+			t:    t,
+			name: "read non-existing value",
+			key:  LogSCMInteractions,
+			u64:  0,
+			err:  ErrNoSuchKey,
+		},
+		{
+			t:    t,
+			name: "reading value returns other error",
+			key:  FlushDNSOnSessionUnlock,
+			u64:  0,
+			err:  errors.New("blah"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handler = &tt
+			value, err := GetUint64(tt.key, 0)
+			if err != nil && tt.err == ErrNoSuchKey {
+				t.Fatalf("got %v error instead of handling ErrNoSuchKey", err)
+			}
+			if value != tt.u64 {
+				t.Fatalf("got value %v instead of expected value %v", value, tt.u64)
+			}
+			if err != tt.err && tt.err != ErrNoSuchKey {
+				t.Fatalf("got error %v instead of expected error %v", err, tt.err)
+			}
+		})
+	}
+}
```

Comment:
Here's an example of splitting out the testHandler parts from the per-test parts

```suggestion
var someOtherError = errors.New("error other than not found")

func setHandlerForTest(tb testing.TB, h Handler) {
	tb.Helper()
	oldHandler := handler
	handler = h
	tb.Cleanup(func() { handler = oldHandler })
}

func TestGetUint64(t *testing.T) {
	tests := []struct{
		name string
		key Key
		handlerValue uint64
		handlerError error
		defaultValue uint64
		wantValue uint64
		wantError error
	}{
		{
			name: "read existing value",
			key:  KeyExpirationNoticeTime,
			handlerValue:  1,
			wantValue: 1,
		},
		{
			name: "read non-existing value",
			key:  LogSCMInteractions,
			handlerError:  ErrNoSuchKey,
			defaultValue: 0,
			wantValue: 0,
		},
		{
			name: "non-zero default value",
			key:  LogSCMInteractions,
			handlerError:  ErrNoSuchKey,
			defaultValue: 5,
			wantValue: 5,
		},
		{
			name: "reading value returns other error",
			key:  FlushDNSOnSessionUnlock,
			defaultValue: 5,
			handlerError:  someOtherError,
			wantValue: 0,
			wantError: someOtherError,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			setHandlerForTest(&testHandler{
				t: t,
				key: tt.key,
				u64: tt.handlerValue,
				err: tt.handlerError,
			})
			value, err := GetUint64(tt.key, 0)
			if err != tt.wantError {
				t.Errorf("err=%q, want %q", err, tt.wantError)
			}
			if value != tt.wantValue {
				t.Errorf("value=%v, want %v", value, tt.wantValue)
			}
		})
	}
}
```